js/main.js

```diff
@@ -105,12 +141,81 @@ function viewSwap(view) {
   }
 }
 
+function populateEntryForm() {
+  const titleInput = document.getElementById('title');
+  const imageInput = document.getElementById('image-url');
+  const notesInput = document.getElementById('user-notes');
+
+  const img = document.querySelector('.display .image').firstChild;
+  img.setAttribute('src', data.editing.image);
+
+  titleInput.value = data.editing.title;
+  imageInput.value = data.editing.image;
+  notesInput.value = data.editing.notes;
+
+  const editTitle = document.querySelector('.new-entry');
+  editTitle.textContent = 'Edit Entry';
+}
+
+function updateOriginalEntry(entry, entryId) {
+  const entries = document.getElementsByTagName('ul');
+  for (const i in entries) {
+    if (entries[i].attributes && parseInt(entries[i].attributes['data-entry-id'].value) === parseInt(entryId)) {
+      entries[i].replaceWith(entry);
+    }
+  }
+}
+
 const entriesLink = document.querySelector('.menu-entries');
 entriesLink.addEventListener('click', function (event) {
   viewSwap('entries');
 });
 
 const newEntry = document.querySelector('.new-entries');
 newEntry.addEventListener('click', function (event) {
+  const editTitle = document.querySelector('.new-entry');
+  editTitle.textContent = 'New Entry';
   viewSwap('entry-form');
 });
+
+$submitForm.addEventListener('submit', function (event) {
+  event.preventDefault();
+  const title = $submitForm.elements.title.value;
+  const image = $submitForm.elements.image.value;
+  const notes = $submitForm.elements.notes.value;
+
+  if (data.editing === null) {
+    const newObject = {
+      title,
+      image,
+      notes,
+      entryId: data.nextEntryId
+    };
+    data.entries.unshift(newObject);
+    data.nextEntryId++;
+    const newEntry = renderEntry(newObject);
+    $ul.prepend(newEntry);
+  } else {
+    const editedEntryId = data.editing.entryId;
+    const editedEntry = {
+      title,
+      image,
+      notes,
+      entryId: editedEntryId
+    };
+    const index = data.entries.findIndex(entry => entry.entryId === editedEntryId);
+    if (index !== -1) {
+      data.entries[index] = editedEntry;
+      const updatedEntry = renderEntry(editedEntry);
+      const liToUpdate = $ul.querySelector(`li[data-entry-id="${editedEntryId}"]`);
+      if (liToUpdate) {
+        liToUpdate.replaceWith(updatedEntry);
+      }
+    }
+    data.editing = null;
+    viewSwap('entries');
+  }
+
+  $img.src = './images/placeholder-image-square.jpg';
+  $submitForm.reset();
+});
```

Comment:
You already have this, you don't need this function again

Reply:
https://github.com/Edward-J-Kim/code-journal/assets/126144853/d0cbcd4f-75e5-4896-9d37-b085f60e7bde

fixed the following comments.